openedx/core/djangoapps/course_groups/tests/test_cohorts.py

```diff
@@ -635,6 +636,30 @@ def test_add_user_to_cohort(self, mock_tracker):
             lambda: cohorts.add_user_to_cohort(first_cohort, "non_existent_username")
         )
 
+    def add_user_to_cohorts_race_condition(self):
+        course_user = UserFactory(username="Username", email="a@b.com")
+        course = modulestore().get_course(self.toy_course_key)
+        CourseEnrollment.enroll(course_user, self.toy_course_key)
+        first_cohort = CohortFactory(course_id=course.id, name="FirstCohort")
+        second_cohort = CohortFactory(course_id=course.id, name="SecondCohort")
+
+        membership1 = CohortMembership(user=course_user, course_user_group=first_cohort)
+        membership2 = CohortMembership(user=course_user, course_user_group=second_cohort)
+
+        with before_after.after('models.CohortMembership._innerRead', membership2.save):
```

Comment:
You could also try pre-pending two dots in front of 'models' like in your import statements above.